module-sys/SystemManager/PowerManager.cpp

```diff
@@ -250,16 +180,37 @@ namespace sys
         cpuGovernor->RemoveSentinel(sentinelName);
     }
 
-    void PowerManager::SetCpuFrequencyRequest(std::string sentinelName,
-                                              bsp::CpuFrequencyMHz request,
-                                              bool permanentBlock)
+    void PowerManager::SetCpuFrequencyRequest(const std::string &sentinelName, bsp::CpuFrequencyMHz request)
+    {
+        cpuGovernor->SetCpuFrequencyRequest(sentinelName, request);
+        // TODO:
+        // - move other UpdateCpuFrequency usages too
+        // - if it's ok to trigger algorithms on leave?
+        auto ret = UpdateCpuFrequency();
```

Comment:
Do we agree that the update will be faster than 100ms? This will affect the FrequencyStepping algorithm with thresholds for that profile

Reply:
Kind of :) The update will be faster than 100ms. 
When the processor is not fully occupied while rising the sentinel it takes an average of 4ms to: send requests and processes. Where it seems that most of the time is taken by context switches but it should be investigated further. 
When the processor is under heavy processing - it may take as long as 17ms (this is what I've seen in logs). 
At the moment we calculate the last %processing each time we update CPU frequency, so it's affecting the 100ms polling time, but it still calculates percent in idle 'from last time asked'.
Saying that: most of the times we set sentinel we are on raising frequency and I didn't see anything worrying about this change. 
In the end: we want to have the upscale of the frequency as fast as possible. It's just wrong to wait up to 100ms (or more) for frequency to rise when we need it as fast as possible ( 1. to process data 2. for sentinel to have a sense which is an upscale processor for further workload)